Add Teleport Connect to Headless docs (#30300)
gravitational/teleport: docs/pages/access-controls/guides/headless.mdx

- Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-client-only) `tsh` binary v12.2+ installed.
- Machines used to approve Headless WebAuthn requests have a Web browser with [WebAuthn support](
  https://developers.yubico.com/WebAuthn/WebAuthn_Browser_Support/) or `tsh` binary v12.2+ installed.

> - Optional: Teleport Connect v13.3.1+ for [seamless headless approval](#optional-teleport-connect).

Reviewer: ```suggestion
- Optional: Teleport Connect v13.3.1+ for [seamless Headless WebAuthn approval](#optional-teleport-connect).
```
We should give this feature a consistent name within this page

  your local terminal.
</Notice>

## Optional: Teleport Connect

> Teleport Connect v13.3.1+ can also be used to approve headless logins.
> Teleport Connect will automatically detect the headless login attempt
> and allow you to approve or cancel the request.
> 
> <Figure width="700">
> ![Headless Confirmation](../../../img/headless/confirmation.png)
> </Figure>
> 
> You will be prompted to tap your MFA key to complete the approval process.
> 
> <Figure width="700">
> ![Headless WebAuthn Approval](../../../img/headless/approval.png)
> </Figure>
> 
> <Notice type="note">
>   This also requires a v13.3.1+ Teleport Auth server.
> </Notice>

Reviewer: ```suggestion
Teleport Connect v13.3.1+ can also be used to approve Headless WebAuthn logins.
Teleport Connect will automatically detect the Headless WebAuthn login attempt
and allow you to approve or cancel the request.

<Figure width="700">
![Headless Confirmation](../../../img/headless/confirmation.png)
</Figure>

You will be prompted to tap your MFA key to complete the approval process.

<Figure width="700">
![Headless WebAuthn Approval](../../../img/headless/approval.png)
</Figure>

<Notice type="note">
  This also requires a v13.3.1+ Teleport Auth Service.
</Notice>
```

So we have consistent product names in the docs.

  your local terminal.
</Notice>

## Optional: Teleport Connect

> Teleport Connect v13.3.1+ can also be used to approve headless logins seamlessly.

Reviewer 2: ```suggestion
Teleport Connect v13.3.1+ can also be used to approve headless logins.
```

> Teleport Connect will automatically detect the headless login attempt and display
> the details in the App for you to `Approve` or `Cancel`.

Reviewer 2: ```suggestion
Teleport Connect will automatically detect the headless login attempt and allow you to approve or cancel the request.
```

<Figure width="700">
![Headless Confirmation](../../../img/headless/confirmation.png)
</Figure>

> When you hit `Approve`, you will be prompted to tap your MFA key to complete the approval

Reviewer 2: I think we're overusing `Code Formatting`. Approve is just a verb, we don't need any special formatting.

```suggestion
You will be prompted to tap your MFA key to complete the approval
```
